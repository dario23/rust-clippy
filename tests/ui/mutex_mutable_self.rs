#![allow(clippy::all)]
#![warn(clippy::mutex_mutable_self)]
#![allow(unused)]

use std::sync::Mutex;


fn get_value_mut(m: &mut Mutex<u32>) -> u32 {
    *m.lock().unwrap()
}

fn get_value(m: &Mutex<u32>) -> u32 {
    *m.lock().unwrap()
}

fn mk_mutex() -> Mutex<u32> {
    Mutex::new(10)
}

fn main() {
    let mut m = Mutex::new(42);
    *m.lock().unwrap() = 64;

    let _ = get_value(&m);
    let _ = get_value_mut(&mut m);

    let _ = mk_mutex().lock().unwrap();
}
