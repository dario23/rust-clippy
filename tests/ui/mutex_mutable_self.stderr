error: exclusive mutex variable
  --> $DIR/mutex_mutable_self.rs:9:6
   |
LL |     *m.lock().unwrap()
   |      ^^^^^^^^
   |
   = note: `-D clippy::mutex-mutable-self` implied by `-D warnings`
   = help: consider `Mutex::get_mut` instead

error: aborting due to previous error

